lib/nexpose/error.rb

```diff
@@ -15,9 +15,9 @@ def to_s
   end
 
   class AuthenticationFailed < APIError
-    def initialize(req)
+    def initialize(req, reason = nil)
```

Comment:
Since we're actually passing in the exception instance in as "reason", how about `cause`? Then we can assign `@reason` to `cause.message || 'Login Failed'`.